File: 0242-valid-anagram/solution.cs
public class Solution {
    public bool IsAnagram(string s, string t) {

        if(s.Length != t.Length) return false; 
        if(s == t) return true;       
        Dictionary<char, int> letters = new Dictionary<char, int>();
        Dictionary<char, int> letters2 = new Dictionary<char, int>();
        
        for(int i=0; i<s.Length; i++){
            letters[s[i]] = 1 + (letters.ContainsKey(s[i]) ? letters[s[i]] : 0);  
            letters2[t[i]] = 1 + (letters2.ContainsKey(t[i]) ? letters2[t[i]] : 0);
        }
        return letters.OrderBy(kv => kv.Key).SequenceEqual(letters2.OrderBy(kv => kv.Key));
    }
}

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> map = new HashMap<>();
        HashMap<Character, Integer> mapt = new HashMap<>();
        int n = s.length();
        int m = t.length();

        if(s.equals(t)) return true;
        if(m == 0 && n == 0 && (!s.equals(t))) return false; 
        if(m!=n) return false;
        for(int i = 0; i<Math.max(m,n); i++){
            if(i < m){
                if(!map.containsKey(s.charAt(i))) map.put(s.charAt(i), 1);
                else map.put(s.charAt(i), map.get(s.charAt(i))+1);
            }
            if(i < n){
                if(!mapt.containsKey(t.charAt(i))) mapt.put(t.charAt(i), 1);
                else mapt.put(t.charAt(i), mapt.get(t.charAt(i))+1);
            }
        }
        return map.equals(mapt);
    }
}
